### DISCUSSION TITLE
Binary file upload

### DISCUSSION THREAD

**Discussion by ajw1970** [stephan1827/mappView-File-Explorer#14] [Q&A]
Hi Steven,

Thank you for the repositories! We're putting this one into production and found a problem when uploading binary files. We get this error in the logger:

![image](https://user-images.githubusercontent.com/3105267/112475988-6ca2d980-8d3f-11eb-9acd-cb01a97f6860.png)

Do you have any suggestions on where we might focus on troubleshooting this?

Thank you.

**Comment by stephan1827** [comment]
Hi John,
I will try to help you with this but I need a few more information. I assume you downloaded the latest source code. Can you please answer the following questions

- What Automation Studio version do you use?
- What runtime version do you use?
- What PLC do you use?
- What mapp version do you use?

Can you also attach the file you used to generate the error?

Thx

Stephan

**Reply by ajw1970** [reply]
Hi Stephan, thank you.

AS 4.8.3.111
AR 4.82
5PPC2100.BY01-000	1.3.1.0	
mapp View 5.8.1

https://weltyautomation-my.sharepoint.com/:u:/p/john/EaiKA2-swFtBjq1EJTxDIi4BT-gAm07RpDY1bfL4iNk-CQ?e=TSp3Vz

**Comment by stephan1827** [comment]
Hi John,
do you need any more help with this?

Stephan

**Reply by ajw1970** [reply]
Looks like the problem isn't binary file specifically but large file. Smaller ones worked fine.

**Reply by stephan1827** [reply]
That is correct, you have to increase the constant EXPL_UPLOAD_DOWNLOAD_SIZE to the minimum size of the file you are trying to upload. I will check in the next days how to avoid the PLC to crash when the file exceeds the buffer size.

**Reply by stephan1827** [reply]
John, can you please try to download the code from here

https://github.com/br-automation-com/mappView-File-Explorer/archive/refs/heads/master.zip

and give it a try. The upload should now report an error if the file is too large. You have to increase EXPL_UPLOAD_DOWNLOAD_SIZE to upload files that large. Keep in mind that the memory you reserve for upload will be taken from the DRAM. That means if you allow files as big as 100MB your PLC memory will be reduced by that amount.

**Reply by stephan1827** [reply]
I created a new version that will report an error on the browser side. Starting with runtime D4.90 this issue should be fixed on the PLC side.

**Reply by ajw1970** [reply]
> John, can you please try to download the code from here
> 
> https://github.com/br-automation-com/mappView-File-Explorer/archive/refs/heads/master.zip
> 
> and give it a try. The upload should now report an error if the file is too large. You have to increase EXPL_UPLOAD_DOWNLOAD_SIZE to upload files that large. Keep in mind that the memory you reserve for upload will be taken from the DRAM. That means if you allow files as big as 100MB your PLC memory will be reduced by that amount.

Thanks Stephan, we'll give this a try ASAP.

**Reply by ajw1970** [reply]
> I created a new version that will report an error on the browser side. Starting with runtime D4.90 this issue should be fixed on the PLC side.

That's AR D4.90 you're referring to?

**Reply by stephan1827** [reply]
Yes, its not released yet but the AsHTTP library is fixed in that version. At some point I will change the code back to the previous version because it is nicer when the PLC reports this error with a message box. The AsHTTP reports a buffer overrun correct up to 1MB but with 2MB or larger it crashes.
